src/JMS/Serializer/GenericSerializationVisitor.php

```diff
@@ -109,16 +109,18 @@ public function visitArray($data, array $type, Context $context)
         $isList = isset($type['params'][0]) && !isset($type['params'][1]);
 
         foreach ($data as $k => $v) {
-            $v = $this->navigator->accept($v, $this->getElementType($type), $context);
+            $val = $this->navigator->accept($v, $this->getElementType($type), $context);
 
-            if (null === $v && $context->shouldSerializeNull() !== true) {
+            if (null === $val && ($context->shouldSerializeNull() !== true
+                || (is_object($v) && null !== $context->getExclusionStrategy() && $context->getMetadataStack()->isEmpty()))
```

Comment:
can you explain me why `(is_object($v) && null !== $context->getExclusionStrategy() && $context->getMetadataStack()->isEmpty())` is necessary? isnt `(null === $v && $context->shouldSerializeNull() !== true)` enough?

Reply:
In fact, it's the `$context->shouldSerializeNull() !== true` condition the problem because I want `null` values to be serialized (`->setSerializeNull(true)`) but not objects serialized to `null` using an exclusion strategy.

Reply:
But as I said in my comment on the PR, I'm not sure that `is_object($v) && null !== $context->getExclusionStrategy() && $context->getMetadataStack()->isEmpty()` is the right check to do. I came to this condition because I went in the `GraphNavigator` class and saw this https://github.com/schmittjoh/serializer/blob/master/src/JMS/Serializer/GraphNavigator.php#L222-L228
But in case of some refactoring, the condition might be wrong :/

Reply:
To me makes much more sense to not serialize nulls, independently from the reason why they are null (simple nulls or exclusion strategies). 

What do you think

Reply:
In my case it will be complicated because it will break my clients' fronts because they don't check for the key's existence but ony the value :/

--

Maybe it's more a "returned value" issue, which shouldn't be considered as the same `null` value ?

I may be wrong (I don't work a lot with serializers) but in my mind `array(new Object(toSkip), new Object(doNotSkip))` (with an exclusion strategy removing the first object) and `array(new Object(doNotSkip))` should return the same result. FYI it's not totally the case anyway because of the array key: in the first example, the second object will have the key `1` and in the second the key `0` in the json/xml/... result.

--

Just to add my use case, we generate a big tree (no limit of level, depends on the configuration) of objects, do lots of things on it, and update a `display (bool)` property which is used in an exclusion strategy to skip the class (`shouldSkipClass`) with `display` at false and remove the property (`shouldSkipProperty`).
All the objects are very important during the process (even the ones that are supposed to be hidden) but we don't want to return some of them (`display` false) through the API because it's a business requirement.